Set default host name in `ch_http_connection()`

theory opened the pull request:
Otherwise, calls to `clickhouse_raw_query()` with a connection string that omits the host name causes an error. Using "localhost" as the default to match the default specified in the `CREATE FUNCTION` default. Add an empty connection string argument to a call to `clickhouse_raw_query()` in `tests/sql/http.sql` to test it.

Resolves #18.

serprex reviewed (approved):
do we also want to fallback to localhost when `!*host`?

theory committed:
Set default host name in `ch_http_connection()`

Otherwise, calls to `clickhouse_raw_query()` with a connection string
that omits the host name causes an error. Using "localhost" as the
default to match the default specified in the `CREATE FUNCTION` default.
Add an empty connection string argument to a call to
`clickhouse_raw_query()` in `tests/sql/http.sql` to test it.

Resolves #18.
